libkbfs/key_bundle_cache.go

```diff
@@ -24,21 +25,27 @@ type keyBundleCacheKey struct {
 	isWriter    bool
 }
 
-// NewKeyBundleCacheStandard constructs a new KeyBundleCacheStandard with the given
-// cache capacity.
-func NewKeyBundleCacheStandard(capacity int) *KeyBundleCacheStandard {
-	head, err := lru.New(capacity)
-	if err != nil {
-		panic(err.Error())
-	}
-	return &KeyBundleCacheStandard{head}
+func (k keyBundleCacheKey) str() string {
+	return k.tlf.String() + k.bundleIDStr + strconv.FormatBool(k.isWriter)
+}
+
+// NewKeyBundleCacheLRU constructs a new KeyBundleCacheStandard with LRU
+// eviction strategy. The capacity of the cache is set to capacityBytes bytes.
+func NewKeyBundleCacheLRU(capacityBytes int) *KeyBundleCacheStandard {
+	return &KeyBundleCacheStandard{cache.NewLRUEvictedCache(capacityBytes)}
+}
+
+// NewKeyBundleCacheRandom constructs a new KeyBundleCacheStandard with random
+// eviction strategy. The capacity of the cache is set to capacityBytes bytes.
+func NewKeyBundleCacheRandom(capacityBytes int) *KeyBundleCacheStandard {
```

Comment:
Planning on using this for `mdmerkle`, since it always goes through all latest metadata, and if we ever need to evict cache entries, LRU might cause us to always have a cache miss assuming scanning order is deterministic.